File: src/main/java/no/nav/syfo/consumer/ws/InngaaendeJournalConsumer.java
package no.nav.syfo.consumer.ws;

import lombok.extern.slf4j.Slf4j;
import no.nav.syfo.domain.InngaaendeJournal;
import no.nav.tjeneste.virksomhet.inngaaende.journal.v1.*;
import org.springframework.stereotype.Component;

@Component
@Slf4j
public class InngaaendeJournalConsumer {

    private InngaaendeJournalV1 inngaaendeJournalV1;

    public InngaaendeJournalConsumer(InngaaendeJournalV1 inngaaendeJournalV1) {
        this.inngaaendeJournalV1 = inngaaendeJournalV1;
    }

    public InngaaendeJournal hentDokumentId(String journalpostId) {
        try {
            WSInngaaendeJournalpost inngaaendeJournalpost = inngaaendeJournalV1.hentJournalpost(
                    new WSHentJournalpostRequest()
                            .withJournalpostId(journalpostId))
                    .getInngaaendeJournalpost();

            return InngaaendeJournal.builder()
                    .dokumentId(inngaaendeJournalpost.getHoveddokument().getDokumentId())
                    .status(inngaaendeJournalpost.getJournaltilstand().name())
                    .build();


        } catch (HentJournalpostSikkerhetsbegrensning e) {
            log.error("Feil ved henting av journalpost: Sikkerhetsbegrensning!");
            throw new RuntimeException("Feil ved henting av journalpost: Sikkerhetsbegrensning!", e);
        } catch (HentJournalpostJournalpostIkkeInngaaende e) {
            log.error("Feil ved henting av journalpost: Journalpost er ikke inngaaende!");
            throw new RuntimeException("Feil ved henting av journalpost: Journalpost er ikke inngaaende!", e);
        } catch (HentJournalpostJournalpostIkkeFunnet e) {
            log.error("Feil ved henting av journalpost: Journalpost ikke funnet!");
            throw new RuntimeException("Feil ved henting av journalpost: Journalpost ikke funnet!", e);
        } catch (HentJournalpostUgyldigInput e) {
            log.error("Feil ved henting av journalpost: Journalpostid ikke gyldig!");
            throw new RuntimeException("Feil ved henting av journalpost: Journalpostid ikke gyldig!", e);
        }
    }
}

File: src/main/kotlin/no/nav/syfo/consumer/ws/InngaaendeJournalConsumer.kt
package no.nav.syfo.consumer.ws

import log
import no.nav.syfo.domain.InngaaendeJournal
import kotlinx.coroutines.runBlocking
import no.nav.syfo.client.saf.SafJournalpostClient

class InngaaendeJournalConsumer(
    private val safJournalpostClient: SafJournalpostClient,
    ) {

    var log = log()

    fun hentDokumentId(journalpostId: String): InngaaendeJournal {
        return runBlocking {
            val response = safJournalpostClient.getJournalpostMetadata(journalpostId)
            InngaaendeJournal(
                dokumentId = response.journalpost?.dokumenter!![0].dokumentInfoId,
                status = response.journalpost.journalstatus,
                mottattDato = response.journalpost.datoOpprettet
            )
        }
    }
}
